### PR TITLE
Add To Cart Back End feature

### PR DESCRIPTION
**Pull request by SteveMarshall-89**
POST
GET

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -51,6 +51,91 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      FROM "cartItems" AS "c"
+      JOIN "products" AS "p" USING ("productId")
+     WHERE "c"."cartId" = $1;
+     `;
+    db.query(text, value)
+      .then(result => res.json(result.rows[0]));
```
Comment (thebearingedge) [inline server/index.js]: Be sure to add a `.catch()` here

File: server/index.js [hunk 2]
```diff
@@ -51,6 +51,91 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+  `;
+  const parsedId = parseInt(req.body.productId);
+  if (parsedId < 0 || isNaN(parsedId)) {
+    return res.status(400).json({ error: 'Invalid Product ID' });
+  }
+  db.query(text, value)
+    .then(priceResult => {
+      const price = priceResult.rows[0];
+      if (!priceResult.rows[0]) {
+        next(new ClientError('Product not found', 400));
```
Comment (thebearingedge) [inline server/index.js]: Here you need to `throw` the error, because calling next does not stop the promise chain.

File: server/index.js [hunk 3]
```diff
@@ -51,6 +51,91 @@ app.get('/api/products/:productId', (req, res, next) => {
...
+             "p"."name",
+             "p"."shortDescription"
+      FROM   "cartItems" AS "c"
+      JOIN   "products" AS "p" USING ("productId")
+      WHERE  "c"."cartItemId" = $1;
+      `;
+      return db.query(text, value)
+        .then(result => res.status(201).json(result.rows[0]));
+    })
+    .catch();
```
Comment (thebearingedge) [inline server/index.js]: Be sure to pass the `err` to `next()`

```js
.catch(err => next(err))
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 2 above)
Comment (SteveMarshall-89) [reply server/index.js]: Submitted suggested change

File: server/index.js (hunk 1 above)
Comment (SteveMarshall-89) [reply server/index.js]: Caught

File: server/index.js (hunk 3 above)
Comment (SteveMarshall-89) [reply server/index.js]: Added error handler

**Pull request merged**